src/extension.ts

```diff
@@ -68,9 +70,92 @@ export async function activate(context: vscode.ExtensionContext) {
 			.then(document => vscode.window.showTextDocument(document));
 	};
 
+	const outputChannel = vscode.window.createOutputChannel("Roblox TS");
+
+	const statusBarItem = vscode.window.createStatusBarItem(
+		vscode.StatusBarAlignment.Right,
+		500
+	);
+
+	const statusBarDefaultState = () => {
+		statusBarItem.text = "$(debug-start) roblox-ts";
+		statusBarItem.command = "roblox-ts.start";
+	};
+
+	let compilerProcess: childProcess.ChildProcessWithoutNullStreams;
+	const startCompiler = async() => {
+		statusBarItem.text = "$(debug-stop) roblox-ts";
+		statusBarItem.command = "roblox-ts.stop";
+
+		if (!vscode.workspace.workspaceFolders) return showErrorMessage("Not in a workspace");
+
+		outputChannel.appendLine("Starting compiler..");
+		outputChannel.show();
+
+		const commandConfiguration = vscode.workspace.getConfiguration("roblox-ts.command");
+		const parameters = commandConfiguration.get<Array<string>>("parameters", ["-w"]);
+		const workspacePath = vscode.workspace.workspaceFolders[0].uri.fsPath;
+		const options = {
+			cwd: workspacePath.toString(),
+			shell: true
+		};
+
+		const development = commandConfiguration.get("development", false);
+		const compilerCommand = development ? "rbxtsc-dev" : "rbxtsc";
+
+		// Detect if there is a local install
+		const localInstall = path.join(workspacePath, "node_modules", ".bin", "rbxtsc");
+
+		vscode.commands.executeCommand('setContext', 'roblox-ts:compilerActive', true);
+		if (!development && fs.existsSync(localInstall)) {
+			outputChannel.appendLine("Detected local install, using local install instead of global");
+			compilerProcess = childProcess.spawn(localInstall, parameters, options);
+		} else {
+			compilerProcess = childProcess.spawn(compilerCommand, parameters, options);
+		}
+
+		compilerProcess.on("error", error => {
+			const errorMessage = `Error while starting compiler: ${error.message}`;
+			showErrorMessage(errorMessage);
+			outputChannel.appendLine(errorMessage);
+		});
+
+		compilerProcess.stdout.on("data", chunk => outputChannel.append(chunk.toString()));
+
+		compilerProcess.on("exit", exitCode => {
+			vscode.commands.executeCommand('setContext', 'roblox-ts:compilerActive', false);
+
+			if (exitCode) {
+				vscode.window.showErrorMessage("Compiler did not exit successfully.", "Show Output").then(choice => {
+					if (!choice) return;
+
+					outputChannel.show();
+				});
+			}
+
+			outputChannel.appendLine(`Compiler exited with code ${exitCode ?? 0}`);
+			statusBarDefaultState();
+		});
+	};
+
+	const stopCompiler = async() => {
+		outputChannel.appendLine("Stopping compiler..");
+		compilerProcess.kill("SIGTERM");
```

Comment:
`compilerProcess.kill` doesn't work, I believe it's terminating the shell instead of roblox-ts. It may be necessary to use a cross platform library like [tree-kill](https://www.npmjs.com/package/tree-kill) to kill the entire process tree.

Reply:
Interesting, it works for me on Windows - I'll give tree-kill a try though

Reply:
Odd behavior with tree-kill, roblox-ts is exiting with code 1 - I'm not sure why, it probably has to do with the fact that tree-kill is killing the processes with the force flag.

Reply:
> Interesting, it works for me on Windows - I'll give tree-kill a try though

Were you using WSL or anything? I'm on Windows too and it doesn't seem to work.

Reply:
I'm not using WSL, I have it disabled actually.

Reply:
Does it work any better with SIGINT? This is probably more natural than SIGTERM anyways

Reply:
Doesn't appear to have changed, I believe node ignores POSIX signals on Windows.. Not sure why it isn't working on my machine, what version of windows are you on?

Reply:
I'm on Windows 10 20H2 as of right now

Reply:
Weird, I don't know what would be causing it then.

How are you testing that it is being killed? What I'm doing is starting and stopping it several times and then going into a file and saving a change. For me, it says several times that changes were detected showing that none of the processes were killed.

The "Compiler exited" does appear anyways, so if you're using that to determine it is being killed then try what I said above.

Reply:
Oh - I see it now, yea it's still running. It's weird how the exit event is still being triggered - it's most likely as you said with child processes. It may be best to implement an edge case for exit code 1 then